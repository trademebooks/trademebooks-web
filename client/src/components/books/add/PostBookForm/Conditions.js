import React, { useState } from 'react';
import { MDBRow, MDBCol } from 'mdbreact';

import './Conditions.scss';

const Conditions = () => {

  const [condition, setCondition] = useState('');

  let conditions = [
    {
      name: "Poor",
      img: require('../../../../img/condition_icons/Poor_condition_default.png'),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Poor_condition.png")
    },
    {
      name: "Fair",
      img: require("../../../../img/condition_icons/Fair_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Fair_condition.png")
    },
    {
      name: "Good",
      img: require("../../../../img/condition_icons/Good_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Good_condition.png")
    },
    {
      name: "Very Good",
      img: require("../../../../img/condition_icons/Verygood_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Verygood_condition.png")
    },
    {
      name: "Like New",
      img: require("../../../../img/condition_icons/Likenew_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Likenew_condition.png")
    },
  ];

  let conditionsElement = [];

  // eslint-disable-next-line
  conditions.map((cond, i) => {
    conditionsElement.push(
      <MDBCol md="2" sm="12" key={i} className="condition">
        <p className="condition-title">{cond.name}</p>
        <img className="condition-img"
          src={condition === cond.name ? cond.imgSelected : cond.img}
          alt={cond.name} name="condition" cond={cond.name}
          onClick={() => setCondition(cond.name)} />
        <p className="condition-desc">{cond.desc}</p>
      </MDBCol>
    )
  });

  return (
    <>
      <div className="mt-5">
        <MDBRow>
          {conditionsElement}
        </MDBRow>
      </div>
    </>
  );
}

export default Conditions;