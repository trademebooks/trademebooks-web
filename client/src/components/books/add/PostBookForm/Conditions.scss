.condition-img {
  width: 8rem;
  margin: 1rem 0rem;
  cursor: pointer;
}