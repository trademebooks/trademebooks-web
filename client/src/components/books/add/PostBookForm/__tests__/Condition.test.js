import React from 'react'
import renderer from 'react-test-renderer'

import Conditions from '../Conditions'

describe('Component: Conditions', () => {
  const props = {
    book: jest.fn(),
    addBook: jest.fn()
  }

  it('should match the snapshot', () => {
    const ConditionsSnapshot = renderer
      .create(<Conditions {...props} />)
      .toJSON()

    expect(ConditionsSnapshot).toMatchSnapshot()
  })
})
