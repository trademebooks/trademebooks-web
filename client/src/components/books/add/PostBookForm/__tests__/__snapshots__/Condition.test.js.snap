// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Conditions should match the snapshot 1`] = `
<div
  className="mt-5"
>
  <div
    className="row justify-content-center"
    data-test="row"
  />
</div>
`;
